lib/Auth/Process/PerunEntitlement.php

```diff
@@ -99,21 +99,23 @@ public function process(&$request)
 
     private function getEduPersonEntitlement(&$request)
     {
-        if (isset($request['perun']['groups'])) {
-            /** allow IDE hint whisperer
-             * @var model\Group[] $groups
-             */
-            $groups = $request['perun']['groups'];
-        } else {
-            throw new Exception(
+        $eduPersonEntitlement = [];
+
+        if (!isset($request['perun']['groups'])) {
+            Logger::error(
                 'perun:PerunEntitlement: ' .
                 'missing mandatory field \'perun.groups\' in request.' .
                 'Hint: Did you configure PerunIdentity filter before this filter?'
             );
+            return $eduPersonEntitlement;
         }
 
-        $eduPersonEntitlement = [];
+        $groups = $request['perun']['groups'];
+
         foreach ($groups as $group) {
+            $groupName = $group->getUniqueName();
+            $groupName = preg_replace('/^(\w*)\:members$/', '$1', $groupName);
```

Comment:
Won't this match also groups that are not the "members" group we are concerned about?
I think it would be good to add check if the "parentGroupId" is null (as there is no parent group for those we want to match) and maybe also that the name of the group is "members".
It should not happen that this will match any other groups, I would just add those checks just to be sure.
What do you think?

Reply:
I think it is not required because the regex search the string ':members' and the ':'char is not allowed in strig before searched string.